pkg/controller/proposal/controller.go

```diff
@@ -818,3 +820,24 @@ func isModelDataCompatible(pluginDataModels []*gpb.ModelData, targetDataModels [
 	}
 	return true
 }
+
+func doCascadingDelete(changeValues map[string]*configapi.PathValue, configStore map[string]*configapi.PathValue) map[string]*configapi.PathValue {
+	// defining new changeValues map, where we will include old changeValues map and new pathValues to be cascading deleted
+	var updChangeValues = make(map[string]*configapi.PathValue)
+	for prefix, changeValue := range changeValues {
+		// if this pathValue has to be deleted, then we need to search for all children of this pathValue
+		if changeValue.Deleted {
+			for path, value := range configStore {
+				// this condition will be true, when pathValue will find itself in the store,
+				// so there is no need to store it again on line 473
```

Comment:
I did not understand the comment re line 473.
Also consider that a deleted object and a child might already be in the `changeValues` - it should not matter as the map will guarantee uniqueness

Reply:
Indeed, this comment is very confusing. I completely removed it..